projects/ngx-datatable/src/lib/components/body/body.component.ts

```diff
@@ -410,7 +410,7 @@ export class DataTableBodyComponent implements OnInit, OnDestroy {
    */
   updatePage(direction: string): void {
     let offset = this.indexes.first / this.pageSize;
-
+    const scrollOffset = offset;
```

Comment:
Couldn't you just calculate the boolean here instead of creating another temp var?

```suggestion
    const scrollInBetween = !Number.isInteger(offset);
```

Reply:
@kfenner  I had done like that before. but Timo suggested we should put calculation inside if block to avoid calculating all the time so moved it.

Reply:
I was not aware of the side effect in the following if statement. I guess code get cleaner if we move it up here. I agree to @kfenner here. Otherwise it gets kind of complex.

Reply:
Thanks @timowolf . reverted back